src/EmbeddedManifold.jl

```diff
@@ -181,7 +133,7 @@ Return the [`Manifold`](@ref) `N` an [`AbstractEmbeddedManifold`](@ref) is embed
 get_embedding(::AbstractEmbeddedManifold)
 
 @decorator_transparent_function function get_embedding(M::AbstractEmbeddedManifold)
-    return decorated_manifold(M)
+    return M.embedding
```

Comment:
Usually the abstract embedded manifolds do not store this, since the `decorated_manifold` is the one that _is_ the embedding.

Reply:
This is also the only change that breaks things and would be much work to all current `AbstractEmbeddedManifold`s. None of them has this field currently (since its just the parent we have to get).

Reply:
This is the key change of my commit, `AbstractEmbeddedManifold` now decorated the embedded manifold. I believe this makes embedded manifolds more coherent. I probably should restrict this method to `EmbeddedManifold` and then go and fix Manifolds.jl.

With this change, `AbstraceEmbeddedManifold`s are much easier to understand: they just decorate in a sense the embedded manifold, providing `embed` to go to the embedding and `project` to go back.

Reply:
But you then store the embedding twice?
Once `M <: N` (i.e. it acts transparently for some parts to its embedding `N`) _and_ you store the same manifold in `M.embedding`? The only thing that might cause is inconsistencies when those two are different?

Then it really decorates the embedded manifold, then `decorated_manifold` should return the manifold that is decorated (as it does for other decorators, on the syntax level) and here the semantic is that this decorated manifold is the embedding. That was the original modelling and is the one I favour.

Reply:
Before this change `AbstractEmbeddedManifolds` didn't really work for cases where `embed!` wasn't a no-op so some amount of fixes is necessary anyway.

Reply:
I don'T see why it should not work?

Especially I would prefer not to store the embedding twice (and introduce a lot of `.embedding` fields).

Reply:
`EmbeddedManifold` represents points using the representation of embedded manifold, so naturally all `AbstractEmbeddedManifold`s should do the same, so to use methods of the embedding calling `embed!` is actually necessary. Unless we go back to representing points using the representation of the embedding.

Reply:
Now you confuse me. You are writing
1) `AbstractEmbeddedManifolds` use representation of the embdedding but
2) to use methods in the embedding one still has to call `embed!`? That contradicts 1). and so does the next sentence.

I think for the `EmbeddedManifold` your sentence is wrong. The concrete case first just “couples” two manifolds in order to define how to embed one in the other. That's why you have both fields.

For the abstract case: Yes. The manifold uses a representation in the embedding. But this is given by transparency and parent and not by a field `.embedding`.

Reply:
> I don'T see why it should not work?
> 
> Especially I would prefer not to store the embedding twice (and introduce a lot of `.embedding` fields).

There won't be any additional `.embedding` fields :).